lib/freedom_patches/net_http_header.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module NetHTTPHeaderPatch
+  def initialize_http_header(initheader)
+    # If no user-agent is set, set it to the default
+    initheader ||= {}
+    user_agent_key =
+      initheader.keys.find { |key| key.to_s.downcase == "user-agent" } || "User-Agent".to_sym
+    initheader[
+      user_agent_key
+    ] ||= "Discourse/#{Discourse::VERSION::STRING}-#{Discourse.git_version}; +https://www.discourse.org/"
```

Comment:
Might be worth updating the user agent here too:

https://github.com/discourse/discourse/blob/d45c53ed1fd0252d0e6b6274fbc4d2501036c8eb/app/jobs/regular/emit_web_hook_event.rb#L136-L136

And perhaps even extending `lib/discourse.rb` to have a `user_agent` method? Then we can reuse it everywhere. It could pass in an override user agent to cover cases like the above.

Up to you but could also be worth overhauling the onebox user agent to use this, and add the same `discourse/version-git +discourse.org` formatting that you use here:

https://github.com/discourse/discourse/blob/d45c53ed1fd0252d0e6b6274fbc4d2501036c8eb/lib/onebox/helpers.rb#L234-L238

Reply:
Nice find 👍 I didn't know our onebox and web hook is already including user agent strings.